sys/can/pkt.c

```diff
@@ -0,0 +1,287 @@
+/*
```

Comment:
Hm, I don't have a concrete proposal right now, but I really dislike the code duplication with pktbuf here. Can you imagine a way to consolidate some common code?